### ISSUE TITLE
GU0022 should not warn for explicit implementation

### ISSUE THREAD

**Issue by JohanLarsson** [GuOrg/Gu.Analyzers#141]
```cs
namespace RoslynSandbox
{
    interface IFoo
    {
        object Value { get; set; }
    }

    class Foo<T> : IFoo
    {
        public T Value { get; set; }

        object IFoo.Value
        {
            get => this.Value;
            set => this.Value = (T)value;
        }
    }
}
```
